test/unit_pool2d.ml

```diff
@@ -0,0 +1,302 @@
+  Alcotest.(check bool) "fun_forward01" true (To_test_forward.fun01 ())
+let fun_forward02 () =
+  Alcotest.(check bool) "fun_forward02" true (To_test_forward.fun02 ())
+let fun_forward03 () =
+  Alcotest.(check bool) "fun_forward03" true (To_test_forward.fun03 ())
+let fun_forward04 () =
+  Alcotest.(check bool) "fun_forward04" true (To_test_forward.fun04 ())
+let fun_forward05 () =
+  Alcotest.(check bool) "fun_forward05" true (To_test_forward.fun05 ())
+let fun_forward06 () =
+  Alcotest.(check bool) "fun_forward06" true (To_test_forward.fun06 ())
+let fun_forward07 () =
+  Alcotest.(check bool) "fun_forward07" true (To_test_forward.fun07 ())
+let fun_forward08 () =
+  Alcotest.(check bool) "fun_forward08" true (To_test_forward.fun08 ())
+let fun_forward09 () =
+  Alcotest.(check bool) "fun_forward09" true (To_test_forward.fun09 ())
+let fun_forward10 () =
+  Alcotest.(check bool) "fun_forward10" true (To_test_forward.fun10 ())
+let fun_forward11 () =
+  Alcotest.(check bool) "fun_forward11" true (To_test_forward.fun11 ())
+
+let fun_max2d_back00 () =
+  Alcotest.(check bool) "fun_max2d_back00" true (To_test_maxpool2d_back.fun00 ())
+let fun_max2d_back01 () =
+  Alcotest.(check bool) "fun_max2d_back01" true (To_test_maxpool2d_back.fun01 ())
+let fun_max2d_back02 () =
+  Alcotest.(check bool) "fun_max2d_back02" true (To_test_maxpool2d_back.fun02 ())
+let fun_max2d_back03 () =
+  Alcotest.(check bool) "fun_max2d_back03" true (To_test_maxpool2d_back.fun03 ())
+let fun_max2d_back04 () =
+  Alcotest.(check bool) "fun_max2d_back04" true (To_test_maxpool2d_back.fun04 ())
+let fun_max2d_back05 () =
+  Alcotest.(check bool) "fun_max2d_back05" true (To_test_maxpool2d_back.fun05 ())
+
+let fun_avg2d_back00 () =
+  Alcotest.(check bool) "fun_avg2d_back00" true (To_test_avgpool2d_back.fun00 ())
+let fun_avg2d_back01 () =
+  Alcotest.(check bool) "fun_avg2d_back01" true (To_test_avgpool2d_back.fun01 ())
+let fun_avg2d_back02 () =
+  Alcotest.(check bool) "fun_avg2d_back02" true (To_test_avgpool2d_back.fun02 ())
+let fun_avg2d_back03 () =
+  Alcotest.(check bool) "fun_avg2d_back03" true (To_test_avgpool2d_back.fun03 ())
+let fun_avg2d_back04 () =
+  Alcotest.(check bool) "fun_avg2d_back04" true (To_test_avgpool2d_back.fun04 ())
+let fun_avg2d_back05 () =
+  Alcotest.(check bool) "fun_avg2d_back05" true (To_test_avgpool2d_back.fun05 ())
+
+let test_set = [
+  "fun_forward00", `Slow, fun_forward00;
+  "fun_forward01", `Slow, fun_forward01;
+  "fun_forward02", `Slow, fun_forward02;
+  "fun_forward03", `Slow, fun_forward03;
+  "fun_forward04", `Slow, fun_forward04;
+  "fun_forward05", `Slow, fun_forward05;
+  "fun_forward06", `Slow, fun_forward06;
+  "fun_forward07", `Slow, fun_forward07;
+  "fun_forward08", `Slow, fun_forward08;
+  "fun_forward09", `Slow, fun_forward09;
+  "fun_forward10", `Slow, fun_forward10;
+  "fun_forward11", `Slow, fun_forward11;
+
+  "fun_max2d_back00", `Slow, fun_max2d_back00;
+  (* "fun_max2d_back01", `Slow, fun_max2d_back01;
```

Comment:
what happed to these tests, why commented them out? Do they cause failure or what?

Reply:
ok, I saw your explanations in the PR.

Reply:
I guess TF chooses to ignore the padding part? Do other systems such as Theano do the same thing, i.e. ignoring the padding part?

Reply:
Yes, that's what TF does in its implementation. My guess is that Theano does the same, otherwise I was wondering how Keras can give the same result when switching between backends. I'll experiment with Theano/Caffe etc. to see how they deal with pooling backward operations for these cases.

Reply:
OK, is it because it was me who implemented this version in Eigen? Otherwise, eigen should call the same function. But if we are about to merge your code in the near future, then it doesn't make sense to patch the existing implementation in Eigen. We only need to guarantee the C implementation is correct and can pass the test, right?

Reply:
Totally agree. These tests are mainly for the C implementation rather than Eigen. And I think you implemented the max/avg_pool2d_backward operations in Eigen.

Reply:
Haha, ok, then it was my fault then .... I was in a hurry and got everything done probably in just a couple of nights ... yes, let's focus on the c implementation. it is not worth to fix the code we are about to ditch away.